examples/cookbook.cpp

```diff
@@ -581,18 +591,13 @@ SetProperty(JSContext* cx, JS::HandleValue y, JS::HandleValue x)
  * object.
  */
 static bool
-CheckProperty(JSContext* cx, JS::HandleValue y)
+CheckProperty(JSContext* cx, JS::HandleValue y, bool* found)
 {
-  bool found;
-
-  assert(y.isObject());
-  JS::RootedObject yobj(cx, &y.toObject());
-  if (!JS_HasProperty(cx, yobj, "myprop", &found))
+  JS::RootedObject yobj(cx);
+  if (!JS_ValueToObject(cx, y, &yobj))
+    return false;
+  if (!JS_HasProperty(cx, yobj, "myprop", found))
     return false;
-  if (found) {
```

Comment:
Oops, you're right. In actual use, I'd think the caller is a better place to handle a property not being found.

Also, the JS_ValueToObject doesn't make a whole lot of sense here -- you're not going to make "myprop" appear by wrapping in a Number or String object. It would be better off to do the `y.isObject()` check but treat it as if the property did not exist. I just don't like asserting isObject(); it's fine if this is only called from "inner" functions, but it doesn't seem very clear here that this might not be called with user-provided input, and for that it's better to fail gracefully.

Obviously, this is a cookbook and so it doesn't matter that much, but I think error handling is so important when using JSAPI that it should be included everywhere.